### DISCUSSION TITLE
Support for other PSA's/endpoints

### DISCUSSION THREAD

**Discussion by optmsp** [Anteris-Dev/autotask-client#83] [General]
Hi @aidan-casey, I know you are working on a new architecture/release?  Nothing for you to change, but want to float a conversation about supporting endpoints other than just Autotask. Right now in our app we are looking to begin building out support for Manage and other PSAs. We'll need to build an adapter layer to make all this work, including using your lib for Autotask (and we really appreciate your work which is why we'll continue throwing some dev time at features you can't get to, like the recent get/put PR).

Just spitballing to start the conversation, what would be required to allow the lib to be PSA agnostic in supporting other endpoints?

**Comment by aidan-casey** [comment]
So if I am understanding correctly, you are probably talking about something along the lines of:

```php
$autotaskClient = new Client(...);

$psa = new Psa($autotaskClient);

$psa->createTicket(...);
```

Assuming that is correct, I will be honest, that is not really something we are looking to take on supporting at this time. At the time that we created this package, we were using Autotask in-house. We have since moved on, but given the amount of attention this package has gotten, we consider it worthwhile to maintain.

A couple of initial thoughts...

1. Create a GitHub organization for organizing core libraries and PSA adapters (e.g., PSABridge).
2. Specify an adapter interface with everything you might need an adapter to implement (e.g., createContact, createTicket, etc.).
3. Build adapters implementing this interface and utilizing API clients (like this one) as dependencies to make that happen.

Feel free to hit me up, happy to give some guidance if needed, but again, not something I'm looking to take on myself at this time. 🙂

**Reply by optmsp** [reply]
Great input. I may hit you up on some advice if/when I go this route!

**Reply by aidan-casey** [reply]
Feel free to do so!
